auction-app-frontend/src/components/LocationForm/LocationForm.jsx

```diff
@@ -19,12 +20,25 @@ const LocationForm = ({ formData, setFormData, handleFinalSubmit }) => {
     const [paymentInfo, setPaymentInfo] = useState({});
     const [error, setError] = useState(null);
     const [loading, setLoading] = useState(false);
+    const [cardInformation, setCardInformation] = useState([...cardInformationFields]);
 
     const methods = useForm({
         mode: "onBlur", 
         defaultValues: formData
     });
 
+    useEffect(() => {
+        const cardFieldsWithRequiredRule = cardInformationFields.map(field => ({
```

Comment:
Do we have to do this inside the useEffect?

Reply:
i changed it to directly initialize the state now instead of using useEffect